prql-php/src/Compiler.php

```diff
@@ -75,113 +68,179 @@ function __construct(?string $lib_path = null)
-            throw new \InvalidArgumentException("Could not convert RQ.");
+        $ffi_options = $this->options_init($options);
+
+        $res = $this->_libprql->rq_to_sql($rq_json, $out);
+
+        $this->options_destroy($ffi_options);
+
+        return $this->convert_result($res);
+    }
+
+    private function options_init(?Options $options = null) {
+        if ($options === null) {
+            $options = new Options();
+        }
+
+        $ffi_options = $this->_libprql->new("struct Options");
+        $ffi_options->format = $options->format;
+        $ffi_options->signature_comment = $options->signature_comment;
+
+        if (isset($options->target)) {
+            $len = strlen($options->target) + 1;
+            $ffi_options->target = \FFI::new("char[$len]", false);
+            \FFI::memcpy($ffi_options->target, $options->target, $len - 1);
+        }
+
+        return $ffi_options;
+    }
+
+    private function options_destroy($ffi_options) {
+        if (!\FFI::isNull($ffi_options->target)) {
+            \FFI::free($ffi_options->target);
         }
+        unset($ffi_options);
+    }
+
+    private function convert_result($ffi_res): Result {
+        $res = new Result();
+
+        // convert string
+        $res->output = $this->convert_string($ffi_res->output);
+
+        $res->messages = array();
+        for ($i = 0; $i < $ffi_res->messages_len; $i++) {
+            $res->messages[$i] = $this->convert_message($ffi_res->messages[$i]);
+        }
+
+        // free the ffi_result
+        $this->_libprql->result_destroy($ffi_res);
+        return $res;
+    }
+
+    private function convert_message($ffi_msg): Message {
+        $msg = new Message();
+
+        // I'm using numbers here, I cannot find a way to refer to MessageKind.Error
+        if ($ffi_msg->kind == 0) {
+            $msg->kind = MessageKind::Error;
+        } else if ($ffi_msg->kind == 1) {
+            $msg->kind = MessageKind::Warning;
+        } else if ($ffi_msg->kind == 2) {
+            $msg->kind = MessageKind::Lint;
+        }
+
+        $msg->code = $this->convert_nullable_string($ffi_msg->code);
+        $msg->reason = $this->convert_string($ffi_msg->reason);
+        $msg->span = $this->convert_span($ffi_msg->span);
+        $msg->hint = $this->convert_nullable_string($ffi_msg->hint);
+
+        $msg->display = $this->convert_nullable_string($ffi_msg->display);
+        $msg->location = $this->convert_location($ffi_msg->location);
+
+        return $msg;
+    }
+
+    private function convert_span($ffi_ptr): ?Span {
+        if (is_null($ffi_ptr) || \FFI::isNull($ffi_ptr)) {
+            return null;
+        }
+        $span = new Span();
+        $span->start = $ffi_ptr[0]->start;
+        $span->end = $ffi_ptr[0]->end;
+        return $span;
+    }
+
+    private function convert_location($ffi_ptr): ?SourceLocation {
+        if (is_null($ffi_ptr) || \FFI::isNull($ffi_ptr)) {
+            return null;
+        }
```

Comment:
Empty line after if conditions.
```suggestion
        }

```